src/modules/netinstall/PackageModel.cpp

```diff
@@ -0,0 +1,266 @@
+{
+    if ( !index.isValid() )
+    {
+        return QVariant();
+    }
+    PackageTreeItem* item = static_cast<PackageTreeItem*>( index.internalPointer() );
+    if ( index.column() == 0 && role == Qt::CheckStateRole )
+    {
+        return item->isSelected();
+    }
+
+    if ( !item->childCount() ) // package
+    {
+        if ( !item->parentItem()->isHidden() && role == Qt::DisplayRole &&
+                index.column() == 0 )
+            return QVariant( item->packageName() );
+        else if ( role == PackageTreeItem::PackageNameRole )
+            return QVariant( item->packageName() );
+        else
+            return QVariant();
+    }
+
+    if ( item->isHidden() && role == Qt::DisplayRole ) // Hidden group
+    {
+        return QVariant();
+    }
+
+    switch ( role )
+    {
+        case Qt::DisplayRole:
+            return item->data( index.column() );
+        case PackageTreeItem::PreScriptRole:
+            return QVariant( item->preScript() );
+        case PackageTreeItem::PackageNameRole:
+            return QVariant( item->packageName() );
+        case PackageTreeItem::PostScriptRole:
+            return QVariant( item->postScript() );
+        default:
+            return QVariant();
+    }
+}
+
+bool
+PackageModel::setData( const QModelIndex& index, const QVariant& value, int role )
+{
+    if ( role == Qt::CheckStateRole && index.isValid() )
+    {
+        PackageTreeItem* item = static_cast<PackageTreeItem*>( index.internalPointer() );
+        item->setSelected( static_cast<Qt::CheckState>( value.toInt() ) );
+
+        emit dataChanged( this->index(0, 0), index.sibling( index.column(), index.row() + 1 ), 
+                QVector<int>( Qt::CheckStateRole ) );
+    }
+    return true;
+}
+
+Qt::ItemFlags
+PackageModel::flags( const QModelIndex& index ) const
+{
+    if ( !index.isValid() )
+        return 0;
+    if ( index.column() == 0 )
+        return Qt::ItemIsUserCheckable | QAbstractItemModel::flags( index );
+    return QAbstractItemModel::flags( index );
+}
+
+QVariant
+PackageModel::headerData( int section, Qt::Orientation orientation, int role ) const
+{
+    if ( orientation == Qt::Horizontal && role == Qt::DisplayRole )
+        return m_columnHeadings.value( section );
+    return QVariant();
+}
+
+QList<QVariant>
+PackageModel::getPackages( bool isCritical ) const
+{
+    QList<PackageTreeItem*> items = getItemPackages( m_rootItem, isCritical );
+    for ( auto package : m_hiddenItems )
+        items.append( getItemPackages( package, isCritical ) );
+    QList<QVariant> packages;
+    for ( auto item : items )
+    {
+        QMap<QString, QVariant> itemMap;
+        itemMap.insert( "pre-script", item->parentItem()->preScript() ); // Only groups have hooks
+        itemMap.insert( "package", item->packageName() );
+        itemMap.insert( "post-script", item->parentItem()->postScript() );
+        packages.append( QVariant( itemMap ) );
```

Comment:
This conversion to QVariant hides the type you are returning (map from "type" of object to the object name), and it's useless because you always return the same type.

I would prefer returning a list of QMaps with no conversion; if you think the type name is too long you can use typedef, or define a little struct to use in place of the QMaps.